### PR TITLE
Add integration tests for diverting inflation to L2

### PR DESCRIPTION
**Pull request by i-stam**

### CODE REVIEW

**Review by justinjmoses — APPROVED** [review]
Looks good once the tests pass

File: test/integration/dual/notifyRewardAmount.integration.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+const ethers = require('ethers');
```
Comment (ajsantander) [inline test/integration/dual/notifyRewardAmount.integration.js:1]: Would suggest to rename the file to `inflation.integration.js` to be more consistent with the other names.

File: test/integration/dual/notifyRewardAmount.integration.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		});
+
+		describe('when a new distribution is added (bridge)', () => {
+			let escrowBalance;
+
+			before('record values', async () => {
+				escrowBalance = await Synthetix.balanceOf(SynthetixBridgeEscrow.address);
+			});
+
+			before('add a new distribution and mint', async () => {
```
Comment (ajsantander) [inline test/integration/dual/notifyRewardAmount.integration.js:52]: nit: split this before in 2:
1) describe "when a new distribution is added", then before "add a new distribution", then test that its there
2) describe "when mint is called", then before "mint", then test

File: test/integration/dual/notifyRewardAmount.integration.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+						currentFeePeriodRewards.add(rewardsToDeposit)
+					);
+				});
+
+				it('increases the RewardEscrowV2 balance', async () => {
+					assert.bnEqual(
+						await Synthetix.balanceOf(RewardEscrowV2.address),
+						rewardEscrowBalanceL2.add(rewardsToDeposit)
+					);
+				});
```
Comment (ajsantander) [inline test/integration/dual/notifyRewardAmount.integration.js:101]: We should probably check what `rewards.integration.js` does here right?
If so, we should probably try to abstract that into a behavior, so that we don't duplicate test code.

File: test/integration/dual/notifyRewardAmount.integration.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+				RewardsDistribution,
+				Synthetix,
+				SynthetixBridgeEscrow,
+				SynthetixBridgeToOptimism,
+			} = ctx.l1.contracts);
+
+			owner = ctx.l1.users.owner;
+		});
+
+		before('approve if needed', async () => {
```
Comment (ajsantander) [inline test/integration/dual/notifyRewardAmount.integration.js:36]: Why is this needed?

Comment (i-stam) [reply test/integration/dual/notifyRewardAmount.integration.js:36]: not needed, good catch!

File: test/integration/dual/notifyRewardAmount.integration.js (hunk 3 above)
Comment (i-stam) [reply test/integration/dual/notifyRewardAmount.integration.js:101]: this is a good point,  functionality is shared

File: test/integration/dual/notifyRewardAmount.integration.js (hunk 2 above)
Comment (i-stam) [reply test/integration/dual/notifyRewardAmount.integration.js:52]: done

**Review by ajsantander — APPROVED** [review]
